src/components/nav/Nav.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { navStyles } from "@components-styles/nav";
+import { faBars, faX } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useTheme } from "@hooks/useTheme";
+import { Link } from "gatsby";
+
+import Header from "./Header";
+import LanguageSelector from "./LanguageSelector";
+import ThemeSelector from "./ThemeSelector";
+
+import {
+  link,
+  navBarsActivated,
+  navUl,
+  navUlLink,
+} from "@components-styles/nav/nav.module.css";
+import {
+  darkestbeigeBackground,
+  whitebeigeBorder,
+} from "@dark-mode/dark-mode.module.css";
+
+const Nav = () => {
+  const [hideMenu, setHideMenu] = useState(true);
+
+  const { t } = useTranslation();
+
+  const { themeState } = useTheme();
+
+  const url = window.location.href;
+
+  //NAVIGATION STYLES
+
+  navStyles.dark.ul = hideMenu
```

Comment:
To można fajnie wydzielić używająć customowego hooka. Zdzwonimy się i Ci wyjaśnie